### PR TITLE
Ch43589/send data to endpoint

### PR DESCRIPTION
**Pull request by emmettcollings**
### Clubhouse Link

https://app.clubhouse.io/neofinancial/story/43589/send-coverage-data-from-action-to-rest-endpoint

### Description of changes

- action will attempt to make a post request to the input url with coverage info
- coverage difference between branch and base branch is returned from request
- comment made on pr is now better formatted
- types and files reorganized

Screenshots:
coverageComment

### Related PRs

> n/a

### Updated dependencies

`axios` added for making http requests

### Future work

The information posted and returned might need to be changed depending on how we want to handle coverage difference.

### Checklist

- [x] Have you fully tested your PR locally (and integration if necessary)?
~- [ ] Have you updated any affected APIs on Insomnia?~
- [x] Have you written thorough tests for your work?

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -18,8 +18,5 @@ jobs:
       - name: Action test
         uses: ./
         with:
+          post-url: https://74c5d22d5829.ngrok.io/api/coverageDiff
```
Comment (iansu) [inline .github/workflows/test.yml:21]: You probably don't want to commit this.

File: action.yml [hunk 2]
```diff
@@ -2,8 +2,12 @@ name: 'coverage'
 description: 'Collects test coverage info'
 author: 'Neo Financial Engineering'
 inputs:
+  post-url:
```
Comment (iansu) [inline action.yml:5]: I'd call this something like `coverageEndpoint`

File: src/get-coverage.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
+import * as lcov from 'lcov-parse';
+import { promisify } from 'util';
+
+import { CommentData } from './types';
+
+const lcovParse = promisify(lcov);
+
+const getCoverage = async (path: string): Promise<CommentData> => {
+  try {
+    const com_data = {
```
Comment (iansu) [inline src/get-coverage.ts:10]: Variable names should always be camelcased.

File: src/get-coverage.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import * as lcov from 'lcov-parse';
```
Comment (iansu) [inline src/get-coverage.ts:1]: You don't need the `* as` part

**Review by iansu — CHANGES_REQUESTED** [review]
This looks good! Just a couple formatting changes. You also need to exclude the `build` directory so those files don't get committed.

File: src/get-coverage.ts (hunk 4 above)
Comment (emmettcollings) [reply src/get-coverage.ts:1]: When I changed this to `import lcov from 'lcov-parse';` I get this error when trying to run the tests:

```
  ● Test suite failed to run

    TypeError [ERR_INVALID_ARG_TYPE]: The "original" argument must be of type Function. Received type undefined

      4 | import { CommentData } from './types';
      5 |
    > 6 | const lcovParse = promisify(lcov);
        |                   ^
      7 |
      8 | const getCoverage = async (path: string): Promise<CommentData> => {
      9 |   try {

      at Object.<anonymous> (src/get-coverage.ts:6:19)
      at Object.<anonymous> (test/lcov.test.ts:1:1)
```
Not really sure what is getting broken here or what excluding the `* as` is doing differently.

**Review by leah-is-offline — DISMISSED** [review]
looks good

File: src/get-coverage.ts (hunk 4 above)
Comment (iansu) [reply src/get-coverage.ts:1]: Weird. Okay, just leave it as is then.

File: src/types.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  lines: CoverageInfo;
+  functions: CoverageInfo;
+  branches: CoverageInfo;
+};
+
+export type PRData = {
+  token: string;
+  repo: string;
+  ref: string;
+  base_ref: string;
```
Comment (iansu) [inline src/types.ts:18]: This should also be camelcase.

**Review by iansu — CHANGES_REQUESTED** [review]
One small change and I think this is ready.

**Review by iansu — APPROVED** [review]

**Pull request merged**